core/src/main/java/com/amazon/opendistroforelasticsearch/sql/data/model/ExprValueUtils.java

```diff
@@ -129,6 +129,10 @@ public static ExprValue fromObjectValue(Object o) {
       return stringValue((String) o);
     } else if (o instanceof Float) {
       return floatValue((Float) o);
+    } else if (o instanceof ExprValue) {
+      // since there is no primitive in Java for differentiating TIMESTAMP DATETIME and DATE
+      // we can allow passing a ExprValue that already contains this information
+      return (ExprValue) o;
```

Comment:
Any insights here? @penghuo

Reply:
I'll wait for a few more comment/suggestions before making changes to the rest.
Thank you @chloe-zh for the comments!